impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -173,8 +132,87 @@ private KafkaConfig buildBrokerConfig(KafkaClusterConfig.ConfigHolder c, Path te
 
     @Override
     public void start() {
-        if (zooFactory != null) {
+        // kraft mode: per-broker: 1 external port + 1 inter-broker port + 1 controller port + 1 anon port
```

Comment:
Moved as part of an attempt to re-bind ports in the event of a failure which didn't really work. 

Saving the actual work of constructing servers etc until start is called has a slight appeal so I've left it here, however happy to move it back if there are opinions that think it should be done earlier.